demo/src/routes/menu/Versions.svelte

```diff
@@ -0,0 +1,86 @@
+<script lang="ts">
+	import {get, writable} from '@amadeus-it-group/tansu';
+	import type {ReadableSignal, Writable} from '@amadeus-it-group/tansu';
+	import type {Directive} from '@agnos-ui/svelte/types';
+	import {createHasFocus} from '@agnos-ui/svelte/services/focustrack';
+	import {onMount} from 'svelte';
+	import {page} from '$app/stores';
+	import {pathToRoot$, selectedFramework$} from '$lib/stores';
+
+	interface Version {
+		folder: string;
+		version: string;
+	}
+
+	const versionsParsed: Version[] = [{folder: 'latest', version: 'Latest'}];
+
+	onMount(() => {
+		fetch('https://amadeusitgroup.github.io/AgnosUI/versions.json').then((res) => {
+			res.json().then((result) => {
+				versionsParsed.push(...result);
+			});
+		});
+		({hasFocus$, directive} = createHasFocus());
+	});
+
+	const currentVersion$: Writable<Version> = writable(versionsParsed[0]);
+	const open$ = writable(false);
+
+	let hasFocus$: ReadableSignal<boolean>;
+	let directive: Directive;
+	$: $open$ = $hasFocus$;
+
+	function giveFocus(el: HTMLAnchorElement, index: number) {
+		if (index === 0) {
+			el.focus();
+		}
+	}
+
+	function getRedirectionURL(version: string) {
```

Comment:
Instead of having this function, you could create a reactive list depending on versions and the page store.
Something like this:
```svelte
<script>
  $: versionsWithUrl = versions.map((version) => ({
    ...version,
    url:
      version.version === 'PREVIEW' || ($page.data['since'] && valid($page.data['since']) && lte($page.data['since'], version.version))
        ? $page.url.pathname.replace(`/${$currentVersion$.folder}/`, `/${version.folder}/`)
        : `${$pathToRoot$.replace($currentVersion$.folder, version.folder)}docs/${$selectedFramework$}/getting-started/introduction`,
  }));
</script>
```